### ISSUE TITLE
Accessibility: improve support for caret configuration

### ISSUE THREAD

**Issue by LiberalArtist** [racket/gui#206]
This issue was first reported on the [mailing list](https://groups.google.com/g/racket-users/c/v7zLoTnDwE0/m/pT5MZSqLCAAJ): some people find a blinking caret either an annoyance or an accessibility issue.

Robby’s https://github.com/racket/gui/commit/1c4e78efc43036aabf77a7a1a313f19e23131b7e was a first step toward addressing this by creating a preference (`'framework:caret-blink-disable?`) to turn blinking off.

As noted, it would be nice to look at the system-level preference when available. The issue https://github.com/electron/electron/issues/10668 seems to have some useful notes.

Beyond disabling blinking, it looks like all platforms support configuring the speed of caret blinking, and Windows and GTK seem to make the width and color of the caret configurable as well. Probably Racket should, too.

**Comment by rfindler** [comment]
Since the code that actually does the drawing, etc., isn't in the framework, but is in wxme, the changes would have to be there for those parts.

**Comment by LiberalArtist** [comment]
I haven't had a chance to test https://github.com/racket/gui/commit/1c4e78efc43036aabf77a7a1a313f19e23131b7e yet, but IIUC making the change at the level of `text:basic-mixin` wouldn't affect, for example, `text-field%`. I haven't personally wanted to configure things this way, but it seems like users who want non-blinking want it in fields, too.

**Comment by rfindler** [comment]
Yes, that's true. I put it in the framework because the preferences are accessible there. But you're right, it is better at the text% level.

**Comment by rfindler** [comment]
Probably I should take it out before the release because of this!

**Comment by loafofpiecrust** [comment]
I'm not sure if this fits here, but I would love to be able to configure the _type_ of caret used in an editor. In emacs, for example, the caret can be one of these types `'(box underline hollow line)`. The most flexible option might be a method for caret drawing that I can override. GTK doesn't seem to support such a thing directly, but if it's possible that would be fantastic. I've been digging around a bit in the wxme code to figure this out, but there are virtually no comments in the code, making it very difficult to decipher.

**Comment by rfindler** [comment]
@loafofpiecrust the editor is actually implemented completely at the Racket level, not using gtk's (or other platform's) editors. Changing the drawing of the insertion caret might be challenging as it is one of the parts of the `text%` that's got performance concerns currently.

**Comment by LiberalArtist** [comment]
> I'm not sure if this fits here, but I would love to be able to configure the _type_ of caret used in an editor.

Related: https://github.com/racket/drracket/issues/667
